### DISCUSSION TITLE
SyncVar OnChange callback multiple parameter options

### DISCUSSION THREAD

**Discussion by vojtechmusilek** [FirstGearGames/FishNet#250] [Feature Request]
Current callback is required to have 3 parameters:
`OnChange method for ... must contain 3 parameters ...`

### Notes
All variants prefer previous value over the next value because you can assess the next value in the field.

In my case the most used would be 0 parameters, so that is the one with highest priority.
Depending on how difficult would it be, the 1 and 2 could be also useful for someone.

### Proposal
(examples with int)
- 3 parameters: `(int prev, int next, bool asServer)`
  - This is the only one already implemented.
- 2 parameters: `(int prev, bool asServer)`
  - When you need see the previous variable and make asServer checks
- 1 parameter: `(bool asServer)`
  - When you only need to make asServer checks
- 1 parameter: `(int prev)`
  - When you need see the previous variable
  - This one is debatable, just added it to make all the combinations possible
  - More preferable 1 parameter would be the one above with boolean
- 0 parameters: `()`
  - When you don't need anything

**Comment by FirstGearGames** [comment]
I considered adding this in at first but that would mean every synctype would need different callbacks. Things suddenly become more complicated because syncvars are done at codegen, but others via user code. I can see how the options would be a little more user friendly but for the amount and risk required to accommodate all synctypes I don't believe it's worth putting in at this time.

Sorry, but for now it stays as is.
